src/pathfora.js

```diff
@@ -2830,20 +2838,25 @@
 
       // if we have the recommendation response cached in session storage
       // use that instead of making a new API request
-      var storedRec = sessionStorage.getItem('PathforaRecommend_' + id);
+      var storedRec = sessionStorage.getItem(PREFIX_REC + id);
 
       if (typeof storedRec === 'string' && params.visited !== false) {
-        var rec = JSON.parse(storedRec);
+        var rec;
+
+        try {
+          rec = JSON.parse(storedRec);
+        } catch (e) {
+          console.warn('Could not parse json stored response:' + e);
+        }
 
         if (rec.data) {
```

Comment:
need to: `if(rec && rec.data){` here if we are going for bulletproof. in the event the json parse fails rec would be undefined and this would throw an error.

```
var rec;
try {
  rec = JSON.prase('invalid');
} catch (e) {
  console.warn('Could not parse json stored response:' + e);
}

console.log(rec.data);
```